lib/assertions.js

```diff
@@ -1492,7 +1486,7 @@ module.exports = expect => {
           const valueKey = valueType.valueForKey(value, keyInValue);
           const valueKeyType = expect.findTypeOf(valueKey);
 
-          if (valueKeyType.is('function')) {
+          if (valueKeyType.is('expect.it')) {
             return valueKey(subjectKey);
```

Comment:
we should probably pass the context here.

```js
expect.context.thisObject = subject.
return valueKey(subjectKey, expect.context);
```

Reply:
@sunesimonsen good shout - I can see about constructing a failing test case and apply the fix a bit later.